packages/worker/src/activity.ts

```diff
@@ -38,14 +38,18 @@ export class Activity {
     inbound: ActivityInboundCallsInterceptor[];
     outbound: ActivityOutboundCallsInterceptor[];
   };
+  /** Logger bound to `logSource: worker`, with metadata from this Activity */
+  private readonly logger;
 
   constructor(
     public readonly info: Info,
     public readonly fn: ActivityFunction<any[], any> | undefined,
     public readonly dataConverter: LoadedDataConverter,
     public readonly heartbeatCallback: Context['heartbeat'],
+    workerLogger: Logger,
     interceptors: ActivityInterceptorsFactory[]
   ) {
+    this.logger = withMetadata(workerLogger, () => this.getLogAttributes());
```

Comment:
Put a comment explaining this is expected to be logged as a "worker" or, even better (IMHO), use the activity context logger here too.

Reply:
>even better (IMHO), use the activity context logger here too.

This is exactly what we want to avoid. Users want to be able to distinguish messages emitted by the SDK vs their own code. But will add comment to make this more obvious.

Reply:
Renamed the property to `this.workerLogger`, and added more detailed comment.

Reply:
It's debatable because these logs have all of the metadata that the user gets to control.
In my mind I always saw these logs that were previously in the interceptors associated with an activity or workflow.